libp2p/handler_test.go

```diff
@@ -0,0 +1,68 @@
+package libp2p
+
+import (
+	"context"
+	"math"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	libp2p "github.com/libp2p/go-libp2p"
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	net "github.com/libp2p/go-libp2p-net"
+	"github.com/status-im/whisper/whisperv6"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	proto = "/shh/6.0"
+)
+
+func TestWhisperOverLibp2p(t *testing.T) {
+	rng := rand.New(rand.NewSource(1))
+	priv, _, err := crypto.GenerateKeyPairWithReader(crypto.Secp256k1, 0, rng)
+	require.NoError(t, err)
+	h, err := libp2p.New(context.Background(),
+		libp2p.ListenAddrStrings("/ip4/127.0.0.1/tcp/7777"),
+		libp2p.Identity(priv),
+	)
+	require.NoError(t, err)
+	w := whisperv6.New(&whisperv6.Config{
+		MinimumAcceptedPOW: 0,
+		MaxMessageSize:     whisperv6.DefaultMaxMessageSize,
+	})
+	h.SetStreamHandler(proto, func(s net.Stream) {
+		if err := Handle(w, s, 0, 0); err != nil {
+			log.Error("Connection was closed with error", "peer", s.Conn().RemotePeer(), "error", err)
+		}
+	})
+	priv, _, err = crypto.GenerateKeyPairWithReader(crypto.Secp256k1, 0, rng)
+	require.NoError(t, err)
+	c, err := libp2p.New(context.Background(),
+		libp2p.Identity(priv),
+	)
+	require.NoError(t, err)
+	c.Peerstore().AddAddr(h.ID(), h.Addrs()[1], 30*time.Second)
+	s, err := c.NewStream(context.TODO(), h.ID(), proto)
+	require.NoError(t, err)
+	msgs := NewStream(s, 2*time.Second, 0)
```

Comment:
This is a bit confusing. One stream is created and then a new one. Maybe we can call it `NewProxyStream` or `NewRLPProxyStream()`.